### PR TITLE
GH-1164: Update Password e2e tests.

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-1164

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      user = helper.registerUser();
+      helper.signOutUser();
+    });
+
+    beforeEach(() => {
+      cy.visit('/');
+      helper.loginWithCredentials(user.email, user.password);
+      cy.selectUserMenuOption('Password');
+      cy.url().should('contain', PAGE_URL_UPDATE_PASSWORD);
+      cy.title().should('eq', PAGE_TITLE_UPDATE_PASSWORD);
```
Comment (znikola) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts]: Why do you need to assert  in `beforeEach`?

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      cy.url().should('contain', PAGE_URL_LOGIN);
+    });
+  });
+
+  describe('when a user is authenticated', () => {
+    let user: any;
+    const newPassword = 'newPassword123!';
+
+    before(() => {
+      cy.visit('/');
```
Comment (znikola) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts]: Is there a reason to repeat the same step from `beforeEach`?

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts]: Then Password item in the My Account menu is one of the feature we need to test.  Moreover, if the test setup fails, we need the early assertion to reduce investigation time.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts (hunk 2 above)
Comment (plabadie) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/update-password.e2e-spec.ts]: The before function runs only once in the beginning and `cy.visit('/');` is required to load the app.

**Pull request merged**